optaplanner-core/src/main/java/org/optaplanner/core/config/solver/SolverConfigs.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.config.solver;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.io.UnsupportedEncodingException;
+import java.util.Objects;
+
+import com.thoughtworks.xstream.XStream;
+import com.thoughtworks.xstream.converters.ConversionException;
+import org.optaplanner.core.api.solver.SolverFactory;
+import org.optaplanner.core.impl.solver.io.XStreamConfigReader;
+
+public class SolverConfigs {
```

Comment:
Instead of throwing these create methods on SolverConfig, I decided to create a SolverConfigs class.
Similar to java.util.Collections, Streams, etc

However, *this is inconsistent* with SolverFactory.create*() and BenchmarkFactory.create*(). We can start deprecating those and moving them too, though...

Wdyt?

Reply:
You make a good point. I was really in doubt on this. Internal consistency trumps JDK consistency?

@yurloc @rsynek Wdyt?

It's pretty easy to change for me btw before merging.